### PR TITLE
Layers can select data from multiple collections in a source

### PR DESCRIPTION
**Pull request by blair1618**

### CODE REVIEW

File: core/src/scene/sceneLoader.cpp [hunk 1]
```diff
@@ -1228,15 +1228,30 @@ void SceneLoader::loadLayers(Node layers, Scene& scene, TileManager& tileManager
...
+                source = data_source.Scalar();
+            }
+        }
+
+        std::vector<std::string> collections;
+        if (Node data_layer = data["layer"]) {
+            if (data_layer.IsScalar()) {
+                collections.push_back(data_layer.Scalar());
+            } else if (data_layer.IsSequence()) {
+                collections = data_layer.as<std::vector<std::string>>();
```
Comment (blair1618) [inline core/src/scene/sceneLoader.cpp]: Just want to point out how cool it is that yaml-cpp supports STL types like this, we should use this more often

Comment (tallytalwar) [inline core/src/scene/sceneLoader.cpp]: This is nice. Didn't know it provides default implementations for stl container conversions.

**Comment by tallytalwar** [comment]
Looks good to me. Once couple of questions are answered/fixed can be merged.

File: core/src/scene/dataLayer.cpp [hunk 2]
```diff
@@ -2,9 +2,9 @@
 
 namespace Tangram {
 
-DataLayer::DataLayer(SceneLayer _layer, const std::string& _source, const std::string& _collection) :
+DataLayer::DataLayer(SceneLayer _layer, const std::string& _source, const std::vector<std::string>& _collections) :
     SceneLayer(_layer),
     m_source(_source),
-    m_collection(_collection) {}
+    m_collections(_collections) {}
```
Comment (tallytalwar) [inline core/src/scene/dataLayer.cpp]: We can do a `std::move` here.

P.S. github app does not respect where you comment :(.

File: core/src/scene/sceneLoader.cpp [hunk 3]
```diff
@@ -1228,15 +1228,30 @@ void SceneLoader::loadLayers(Node layers, Scene& scene, TileManager& tileManager
...
         Node data = layer.second["data"];
-        Node data_layer = data["layer"];
-        Node data_source = data["source"];
 
-        std::string collection = data_layer ? data_layer.as<std::string>() : name;
-        std::string source = data_source ? data_source.as<std::string>() : "";
+        std::string source;
+        if (Node data_source = data["source"]) {
+            if (data_source.IsScalar()) {
+                source = data_source.Scalar();
```
Comment (tallytalwar) [inline core/src/scene/sceneLoader.cpp]: a log message should be good here, if `data_source` is not scalar.

File: core/src/scene/dataLayer.cpp (hunk 2 above)
Comment (blair1618) [inline core/src/scene/dataLayer.cpp]: Could, but it's a small, one-time copy

File: core/src/scene/sceneLoader.cpp (hunk 3 above)
Comment (blair1618) [inline core/src/scene/sceneLoader.cpp]: Warnings would be good in many places while loading the style sheet, I could add 3 to 5 different warnings in this block alone, so before peppering log messages everywhere I'd like to think about a more general validation step before reading configuration at all.

**Pull request merged**